From 9319aeccba3ce6063661613106c7b80903310ed6 Mon Sep 17 00:00:00 2001
From: Gustavo Diaz Prado <x0083741@ti.com>
Date: Mon, 21 Mar 2011 18:42:05 -0500
Subject: OMAP4: SGX: UDD: Fix potential OCP DSS error when flushing buffers

This patch fixes sending an incorrect address to the DSS when the
flushing buffers mechanism is triggered in the SGX display driver

Change-Id: I9cb13d25fd5938ef69238c1d16871ac4ceab1687
Signed-off-by: Gustavo Diaz Prado <x0083741@ti.com>
---
 drivers/gpu/pvr/omaplfb/omaplfb_displayclass.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/pvr/omaplfb/omaplfb_displayclass.c b/drivers/gpu/pvr/omaplfb/omaplfb_displayclass.c
index ca9bf3e..0afff54 100644
--- a/drivers/gpu/pvr/omaplfb/omaplfb_displayclass.c
+++ b/drivers/gpu/pvr/omaplfb/omaplfb_displayclass.c
@@ -178,10 +178,8 @@ static void FlushInternalSyncQueue(OMAPLFB_SWAPCHAIN *psSwapChain)
 
 		/* Flip the buffer if it hasn't been flipped */
 		if(psFlipItem->bFlipped == OMAP_FALSE)
-		{
 			OMAPLFBFlip(psSwapChain,
-				(unsigned long)psFlipItem->sSysAddr);
-		}
+				(unsigned long)psFlipItem->sSysAddr->uiAddr);
 
 		/* If the command didn't complete, assume it did */
 		if(psFlipItem->bCmdCompleted == OMAP_FALSE)
-- 
1.6.0.4

